### PR TITLE
Add static host user docs

### PR DESCRIPTION
**Pull request by atburke**
This change adds docs for static host users.

Part of #42712.

### CODE REVIEW

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx [hunk 1]
```diff
@@ -235,6 +237,135 @@ not already exist are created on the host. The `nginxrestarter` user is added to
...
+In this section, you will configure Teleport to create local users independently
+of an SSH session.
+
+### Step 1/3. Configure RBAC
+
+#### Label your server
+
+1. Access the server you enrolled with Teleport before beginning this guide and
+   open the Teleport configuration file, which is `/etc/teleport.yaml` by
+   default.
```
Comment (zmb3) [inline docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:251]: This made me question - do static host users also get created on servers that use the server info resource for labeling?

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx [hunk 2]
```diff
@@ -235,6 +237,135 @@ not already exist are created on the host. The `nginxrestarter` user is added to
...
+Add this content to a file called `nginxrestarter.yaml`:
+
+```yaml
+kind: static_host_user
+metadata:
+  name: nginxrestarter
+spec:
+  matchers:
+      # Use either node_labels or node_labels_expression to select which servers
+      # to create the host user on. Only one is required.
```
Comment (zmb3) [inline docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:309]:
```suggestion
    # Use either node_labels or node_labels_expression to select which servers
    # to create the host user on. Only one is required.
```

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx [hunk 3]
```diff
@@ -235,6 +237,135 @@ not already exist are created on the host. The `nginxrestarter` user is added to
...
+<Admonition type="note">
+
+Deleting the static host user resource does not remove the host user from any
+servers it has been applied to.
+
+</Admonition>
+
+### Step 3/3. Test
+
+Connect to the remote Node via `tsh` and see that the host user was created:
```
Comment (zmb3) [inline docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:356]:
```suggestion
Connect to the remote Node via `tsh` to verify that the host user was created:
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx [hunk 4]
```diff
@@ -235,6 +237,135 @@ not already exist are created on the host. The `nginxrestarter` user is added to
 the `ubuntu`, `nginx`, and `other` groups, as specified in the `host_groups`
 field.
 
+## Static host users
+
+In this section, you will configure Teleport to create local users independently
+of an SSH session.
```
Comment (stevenGravy) [inline docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:243]: This should mention that the SSH service must be >=16.3.0.

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx (hunk 1 above)
Comment (atburke) [reply docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:252]: Yes, any labels that the server heartbeats back to auth are available for static host users.

File: docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx [hunk 5]
```diff
@@ -235,6 +237,138 @@ not already exist are created on the host. The `nginxrestarter` user is added to
...
+
+    # Add additional matchers if needed to configure nginxrestarter differently
+    # on different servers.
+    # - node_labels:
+      # ...
+```
+
+The name of the static host user resource must match the login of the user to
+be created. If your static host user has more than one matcher, a server can't
+be matched by more than one of them.
```
Comment (stevenGravy) [inline docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx:341]: Would this match to the first one in order?  Good if we can say that

Comment (atburke) [reply docs/pages/enroll-resources/server-access/guides/host-user-creation.mdx]: No. If more than one matcher applies to a server, it's an error and the static host user won't be applied at all.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**